src/java.base/share/man/java.md

```diff
@@ -2883,6 +2883,43 @@ Java HotSpot VM.
     expensive operations. Using a lower value will cause heap memory to be
     uncommitted earlier, at the risk of soon having to commit it again.
 
+`-XX:+UseShenandoahGC`
+:   Enables the use of the Shenandoah garbage collector. This is a low pause
+    time, concurrent garbage collector. Its pause times are not proportional to
+    the size of the heap. Shenandoah garbage collector can work with compressed
+    pointers. See `-XX:UseCompressedOops` for further information about
+    compressed pointers.
+
+`-XX:ShenandoahGCMode=`*mode*
+:   Sets the GC mode for Shenandoah GC to use. By default, this option is set
+    to `satb`. Among other things, this defines which barriers are in use.
+    Possible mode values include the following:
+
+    `satb`
+    :   Snapshot-at-the-beginning concurrent GC (three pass mark-evac-update).
+        It is a single generation GC.
+
+    `generational`
+    :   It is also a snapshot-at-the-beginning and concurrent GC, but it is
+        generational. Please see [JEP 404](https://openjdk.org/jeps/404) and
+        [JEP 521](https://openjdk.org/jeps/521) for its advantages and risks.
+
+`-XX:ShenandoahGCHeuristics=`*heuristics*
+:   Sets the heuristics for Shenandoah GC to use. By default, this option is
+    set to `adaptive`. This fine-tunes the GC mode selected, by choosing when
+    to start the GC, how much to process on each cycle, and what other features
+    to automatically enable. Possible heuristics are following:
+
+    `adaptive`
+    :   To maintain the given amount of free heap at all times, even during
+        the GC cycle.
```

Comment:
May be this part (and in `shenandoah_globals.hpp`) could be changed so as not to repeat the description of `adaptive`, by stating that the default setting is adaptive, but replacing the current description of `adaptive` at line 2918 with the text earlier at line 2909-2911.